src/ApiIncrementalChecker.ts

```diff
@@ -160,4 +164,21 @@ export class ApiIncrementalChecker implements IncrementalCheckerInterface {
       allLints.map(this.createNormalizedMessageFromRuleFailure)
     );
   }
+
+  private emitFiles() {
+    try {
+      const program = this.tsIncrementalCompiler.getProgram();
+
+      if (!program) {
+        throw new Error(
+          'Program (a ts.Program object) in emitFiles() for ApiIncrementalChecker.ts not found'
+        );
+      }
+
+      program.emit(); // called here but the emission of files is in fact controlled by the boolean canEmit inside writeFile of CompilerHost
```

Comment:
I don't see any new tests in this PR - please add tests for emitting files :)

Reply:
I am working on this, would you have any more suggestions on tips on how you would like it to be done? :)

Reply:
I am working on this, would you have any more suggestions on tips on how you would like it to be done @piotr-oles ? :)

Reply:
Given the nature of the change I think that integration tests are what we'd be interested in.  Ideally add them to the existing integration test pack.  

For an idea of a test that actually makes changes to the filesystem (as your change would seek to do) you might want to look at the tslint autofix test: https://github.com/Realytics/fork-ts-checker-webpack-plugin/blob/06f565d2b803c6fa2070a59b80fc2ce2660ce30d/test/integration/index.spec.js#L107

Reply:
Thank you John, I will have a look on this.